homeassistant/components/doorbird/config_flow.py

```diff
@@ -90,6 +90,8 @@ def __init__(self):
 
         if macaddress[:6] != DOORBIRD_OUI:
             return self.async_abort(reason="not_doorbird_device")
+        if discovery_info[CONF_HOST].startswith("169.254"):
```

Comment:
We have an is_local helper in `util/network.py`. But 169.254 doesn't seem to be one?

Reply:
Is it local? 169.254.0.0/16 is considered to be self-assigned in general, I think the RFC specifies it as a local-link for auto-configuration or something.

Reply:
axis has the same issue

Reply:
https://en.wikipedia.org/wiki/Link-local_address

Reply:
IPv4 link-local addresses are assigned to address block 169.254.0.0/16 (169.254.0.0 through 169.254.255.255). In IPv6, they are assigned the address block fe80::/10.[2]

Reply:
Then we should update our util function? https://github.com/home-assistant/core/blob/dev/homeassistant/util/network.py

Reply:
Ideally network.py provides

- is_loopback (127.0.0.0/8, ::1/128)
- is_private (fd00::/8, 10.0.0.0/8, 172.16.0.0/12, 192.168.0.0/16)
- is_link_local (169.254.0.0/16)
- is_local (existing functionality for legacy compatibility)

Reply:
I'm thinking merge as-is and I should do that in a followup w/ and update to axis as well?
```
homeassistant/components/axis/config_flow.py:        if discovery_info[CONF_HOST].startswith("169.254"):
```

Reply:
https://github.com/home-assistant/core/pull/33499